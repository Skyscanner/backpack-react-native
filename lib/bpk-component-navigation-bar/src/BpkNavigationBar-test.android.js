/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import BpkNavigationBar from './BpkNavigationBar.android';
import BpkNavigationBarButtonAndroid from './BpkNavigationBarButtonAndroid.android';

describe('android', () => {
  describe('BpkNavigationBar', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Back"
                icon="long-arrow-left"
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom style', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Back"
                icon="long-arrow-left"
                onPress={jest.fn()}
              />
            }
            style={{ zIndex: 5 }}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom theme', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Back"
                icon="long-arrow-left"
                onPress={jest.fn()}
              />
            }
            theme={{
              navigationBarBackgroundColor: 'white',
              navigationBarTintColor: 'red',
              navigationBarDisabledTintColor: 'red',
              navigationBarStatusBarColor: 'transparent',
            }}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with trailing button', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Close"
                icon="close"
                onPress={jest.fn()}
              />
            }
            trailingButton={
              <BpkNavigationBarButtonAndroid
                title="Done"
                icon="tick"
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with a title view', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title={<View testID="title-view" />}
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Back"
                icon="long-arrow-left"
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with a subtitle view', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            subtitleView={<View testID="subtitle-view" />}
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Back"
                icon="long-arrow-left"
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with an icon in the title', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title={{ value: 'Checkout', icon: 'lock', iconPosition: 'leading' }}
            leadingButton={
              <BpkNavigationBarButtonAndroid
                title="Back"
                icon="long-arrow-left"
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('should forward custom props', () => {
    const tree = renderer
      .create(
        <BpkNavigationBar
          title="Backpack"
          testID="bar"
          leadingButton={
            <BpkNavigationBarButtonAndroid
              testID="leading"
              title="Back"
              icon="long-arrow-left"
              onPress={jest.fn()}
            />
          }
          trailingButton={
            <BpkNavigationBarButtonAndroid
              testID="trailing"
              title="Done"
              icon="tick"
              onPress={jest.fn()}
            />
          }
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
