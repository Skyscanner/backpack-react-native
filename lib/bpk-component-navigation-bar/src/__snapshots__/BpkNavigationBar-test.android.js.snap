// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`android BpkNavigationBar should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
              },
              null,
            ],
          ]
        }
      >
        Backpack
      </Text>
    </View>
  </View>
</View>
`;

exports[`android BpkNavigationBar should render correctly with a subtitle view 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
      Object {
        "paddingBottom": 16,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
              },
              null,
            ],
          ]
        }
      >
        Backpack
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      testID="subtitle-view"
    />
  </View>
</View>
`;

exports[`android BpkNavigationBar should render correctly with a title view 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Array [
            null,
            Object {
              "maxHeight": 32,
            },
          ]
        }
        testID="title-view"
      />
    </View>
  </View>
</View>
`;

exports[`android BpkNavigationBar should render correctly with an icon in the title 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "BpkIcon",
              "fontSize": 24,
              "includeFontPadding": false,
            },
            Object {
              "fontSize": 16,
            },
            Array [
              Object {
                "marginEnd": 4,
              },
              Object {
                "color": "rgb(255, 255, 255)",
              },
            ],
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
              },
              Object {
                "lineHeight": 24,
              },
            ],
          ]
        }
      >
        Checkout
      </Text>
    </View>
  </View>
</View>
`;

exports[`android BpkNavigationBar should render correctly with custom style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
      Object {
        "zIndex": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
              },
              null,
            ],
          ]
        }
      >
        Backpack
      </Text>
    </View>
  </View>
</View>
`;

exports[`android BpkNavigationBar should render correctly with custom theme 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872349696,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "red",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "red",
              },
              null,
            ],
          ]
        }
      >
        Backpack
      </Text>
    </View>
  </View>
</View>
`;

exports[`android BpkNavigationBar should render correctly with trailing button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Close"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
              },
              null,
            ],
          ]
        }
      >
        Backpack
      </Text>
    </View>
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Done"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`android should forward custom props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(7, 112, 227)",
        "elevation": 3,
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "width": "100%",
      },
    ]
  }
  testID="bar"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Back"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="leading"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "bottom": 15,
            "end": 72,
            "position": "absolute",
            "start": 72,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgb(17, 18, 54)",
              "fontFamily": "sans-serif",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": -0.4,
              "textAlign": "left",
            },
            Object {
              "fontFamily": "sans-serif-medium",
            },
            Array [
              Object {
                "color": "rgb(255, 255, 255)",
              },
              null,
            ],
          ]
        }
      >
        Backpack
      </Text>
    </View>
    <View
      style={
        Object {
          "borderRadius": 40,
        }
      }
    >
      <View
        accessibilityLabel="Done"
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "borderless": true,
            "color": 872415231,
            "rippleRadius": undefined,
            "type": "RippleAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="trailing"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "rgb(17, 18, 54)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Array [
                Object {
                  "padding": 8,
                },
                Object {
                  "color": "rgb(255, 255, 255)",
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
